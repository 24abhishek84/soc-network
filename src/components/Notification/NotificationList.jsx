import React from 'react';

import { ListGroup, Button, Tabs, Tab } from "react-bootstrap";
import moment from "moment";
import './Notification.scss';

const notificationList = [
  {
    image: "https://gw.alipayobjects.com/zos/rmsportal/ThXAXghbEsBCCSDihZxY.png",
    message: "You received 14 new weekly newspapers",
    timestamp: 1545368400
  },
  {
    image: "https://gw.alipayobjects.com/zos/rmsportal/ThXAXghbEsBCCSDihZxY.png",
    message: "You received 14 new weekly newspapers",
    timestamp: 1545368400
  },
  {
    image: "https://gw.alipayobjects.com/zos/rmsportal/OKJXDXrmkNshAMvwtvhu.png",
    message: "You received 14 new weekly newspapers",
    timestamp: 1545368400
  },
  {
    image: "https://gw.alipayobjects.com/zos/rmsportal/kISTdvpyTAhtGxpovNWd.png",
    message: "You received 14 new weekly newspapers",
    timestamp: 1545368400
  },
  {
    image: "https://gw.alipayobjects.com/zos/rmsportal/GvqBnKhFgObvnSGkDsje.png",
    message: "You received 14 new weekly newspapers",
    timestamp: 1545368400
  },
  {
    image: "https://gw.alipayobjects.com/zos/rmsportal/OKJXDXrmkNshAMvwtvhu.png",
    message: "You received 14 new weekly newspapers",
    timestamp: 1545368400
  },
];

const eventList = [
  {
    image: "https://gw.alipayobjects.com/zos/rmsportal/ThXAXghbEsBCCSDihZxY.png",
    message: "All Events Posted",
    timestamp: 1545368400
  },
  {
    image: "https://gw.alipayobjects.com/zos/rmsportal/ThXAXghbEsBCCSDihZxY.png",
    message: "React Event Posted",
    timestamp: 1545368400
  },
  {
    image: "https://gw.alipayobjects.com/zos/rmsportal/OKJXDXrmkNshAMvwtvhu.png",
    message: "A new meetup at falana complex",
    timestamp: 1545368400
  },
];

const messageList = [
  {
    image: "https://gw.alipayobjects.com/zos/rmsportal/ThXAXghbEsBCCSDihZxY.png",
    message: "You have received new message from Falana Dhikana",
    timestamp: 1545368400
  },
  {
    image: "https://gw.alipayobjects.com/zos/rmsportal/ThXAXghbEsBCCSDihZxY.png",
    message: "You have received new message from Someone",
    timestamp: 1545368400
  },
  {
    image: "https://gw.alipayobjects.com/zos/rmsportal/OKJXDXrmkNshAMvwtvhu.png",
    message: "Someone wants to connect with you",
    timestamp: 1545368400
  },
];

const ListGroupItems = ({ list, type }) => {
  return (
    <React.Fragment>
      <ListGroup className="list-items shadow">
        {list.map((notification, index) => {
          return (
            <ListGroup.Item key={`${type}-${index}`} className={`border-left-0 border-right-0 rounded-0 list-item pro-components-notice-icon-notice-list-item ${index === 0 ? 'border-top-0' : ''}`}>
              <div className="list-item-meta pro-components-notice-icon-notice-list-meta">
                <div className="list-item-meta-avatar">
                  <span className="avatar pro-components-notice-icon-notice-list-avatar avatar-circle avatar-image">
                    <img src={notification.image} alt="" />
                  </span>
                </div>
                <div className="list-item-meta-content">
                  <h4 className="list-item-meta-title">
                    <div className="pro-components-notice-icon-notice-list-title">
                      {notification.message}
                    </div>
                  </h4>
                  <div className="list-item-meta-description">
                    <div>
                      <div className="pro-components-notice-icon-notice-list-description"></div>
                      <div className="pro-components-notice-icon-notice-list-datetime">{moment(new Date(notification.timestamp * 1000)).fromNow()}</div>
                    </div>
                  </div>
                </div>
              </div>
            </ListGroup.Item>
          );
        })}
      </ListGroup>
      {notificationList.length > 0 &&
        <Button variant="primary" className="border-0 rounded-0 clear-all-button pt-2 pb-2 shadow font-size-14 font-weight-bold" block>Clear All</Button>
      }
    </React.Fragment>
  )
}

const NotificationList = () => {
  return (
    <React.Fragment>
      <Tabs defaultActiveKey="notifications" id="uncontrolled-tab-example">
        <Tab tabClassName="notification-tab-item" eventKey="notifications" title={`Notifications (${notificationList.length})`}>
          <ListGroupItems list={notificationList} type="notifications" />
        </Tab>
        <Tab tabClassName="notification-tab-item" eventKey="events" title={`Events (${eventList.length})`}>
          <ListGroupItems list={eventList} type="events" />
        </Tab>
        <Tab tabClassName="notification-tab-item" eventKey="messages" title={`Messages (${messageList.length})`}>
          <ListGroupItems list={messageList} type="messages" />
        </Tab>
      </Tabs>
    </React.Fragment>
  )
}

export default NotificationList
