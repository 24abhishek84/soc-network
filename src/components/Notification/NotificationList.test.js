import React from 'react';
import { shallow } from 'enzyme';
import NotificationList from "./NotificationList";
import renderer from 'react-test-renderer';

describe('Notification List Component', () => {
  let wrapper, snapshot;
  wrapper = shallow(<NotificationList />);
  snapshot = renderer.create(<NotificationList />).toJSON();

  // it('Should render without errors', () => {
  //   const component = findByTestAtrr(wrapper, 'appComponent');
  //   expect(component.length).toBe(1);
  // });

  test('it matches the snapshot', () => {
    expect(snapshot).toMatchSnapshot();
  });
});
