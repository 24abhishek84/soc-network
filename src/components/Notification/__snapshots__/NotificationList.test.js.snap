// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification List Component it matches the snapshot 1`] = `
Array [
  <nav
    className="nav nav-tabs"
    onKeyDown={[Function]}
    role="tablist"
  >
    <a
      aria-controls="uncontrolled-tab-example-tabpane-notifications"
      aria-selected={true}
      className="notification-tab-item nav-item nav-link active"
      data-rb-event-key="notifications"
      href="#"
      id="uncontrolled-tab-example-tab-notifications"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="tab"
    >
      Notifications (6)
    </a>
    <a
      aria-controls="uncontrolled-tab-example-tabpane-events"
      aria-selected={false}
      className="notification-tab-item nav-item nav-link"
      data-rb-event-key="events"
      href="#"
      id="uncontrolled-tab-example-tab-events"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="tab"
      tabIndex={-1}
    >
      Events (3)
    </a>
    <a
      aria-controls="uncontrolled-tab-example-tabpane-messages"
      aria-selected={false}
      className="notification-tab-item nav-item nav-link"
      data-rb-event-key="messages"
      href="#"
      id="uncontrolled-tab-example-tab-messages"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="tab"
      tabIndex={-1}
    >
      Messages (3)
    </a>
  </nav>,
  <div
    className="tab-content"
  >
    <div
      aria-hidden={false}
      aria-labelledby="uncontrolled-tab-example-tab-notifications"
      className="fade tab-pane active show"
      id="uncontrolled-tab-example-tabpane-notifications"
      role="tabpanel"
    >
      <div
        className="list-items shadow list-group"
        onKeyDown={[Function]}
      >
        <div
          className="border-left-0 border-right-0 rounded-0 list-item pro-components-notice-icon-notice-list-item border-top-0 list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <div
            className="list-item-meta pro-components-notice-icon-notice-list-meta"
          >
            <div
              className="list-item-meta-avatar"
            >
              <span
                className="avatar pro-components-notice-icon-notice-list-avatar avatar-circle avatar-image"
              >
                <img
                  alt=""
                  src="https://gw.alipayobjects.com/zos/rmsportal/ThXAXghbEsBCCSDihZxY.png"
                />
              </span>
            </div>
            <div
              className="list-item-meta-content"
            >
              <h4
                className="list-item-meta-title"
              >
                <div
                  className="pro-components-notice-icon-notice-list-title"
                >
                  You received 14 new weekly newspapers
                </div>
              </h4>
              <div
                className="list-item-meta-description"
              >
                <div>
                  <div
                    className="pro-components-notice-icon-notice-list-description"
                  />
                  <div
                    className="pro-components-notice-icon-notice-list-datetime"
                  >
                    10 months ago
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="border-left-0 border-right-0 rounded-0 list-item pro-components-notice-icon-notice-list-item  list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <div
            className="list-item-meta pro-components-notice-icon-notice-list-meta"
          >
            <div
              className="list-item-meta-avatar"
            >
              <span
                className="avatar pro-components-notice-icon-notice-list-avatar avatar-circle avatar-image"
              >
                <img
                  alt=""
                  src="https://gw.alipayobjects.com/zos/rmsportal/ThXAXghbEsBCCSDihZxY.png"
                />
              </span>
            </div>
            <div
              className="list-item-meta-content"
            >
              <h4
                className="list-item-meta-title"
              >
                <div
                  className="pro-components-notice-icon-notice-list-title"
                >
                  You received 14 new weekly newspapers
                </div>
              </h4>
              <div
                className="list-item-meta-description"
              >
                <div>
                  <div
                    className="pro-components-notice-icon-notice-list-description"
                  />
                  <div
                    className="pro-components-notice-icon-notice-list-datetime"
                  >
                    10 months ago
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="border-left-0 border-right-0 rounded-0 list-item pro-components-notice-icon-notice-list-item  list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <div
            className="list-item-meta pro-components-notice-icon-notice-list-meta"
          >
            <div
              className="list-item-meta-avatar"
            >
              <span
                className="avatar pro-components-notice-icon-notice-list-avatar avatar-circle avatar-image"
              >
                <img
                  alt=""
                  src="https://gw.alipayobjects.com/zos/rmsportal/OKJXDXrmkNshAMvwtvhu.png"
                />
              </span>
            </div>
            <div
              className="list-item-meta-content"
            >
              <h4
                className="list-item-meta-title"
              >
                <div
                  className="pro-components-notice-icon-notice-list-title"
                >
                  You received 14 new weekly newspapers
                </div>
              </h4>
              <div
                className="list-item-meta-description"
              >
                <div>
                  <div
                    className="pro-components-notice-icon-notice-list-description"
                  />
                  <div
                    className="pro-components-notice-icon-notice-list-datetime"
                  >
                    10 months ago
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="border-left-0 border-right-0 rounded-0 list-item pro-components-notice-icon-notice-list-item  list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <div
            className="list-item-meta pro-components-notice-icon-notice-list-meta"
          >
            <div
              className="list-item-meta-avatar"
            >
              <span
                className="avatar pro-components-notice-icon-notice-list-avatar avatar-circle avatar-image"
              >
                <img
                  alt=""
                  src="https://gw.alipayobjects.com/zos/rmsportal/kISTdvpyTAhtGxpovNWd.png"
                />
              </span>
            </div>
            <div
              className="list-item-meta-content"
            >
              <h4
                className="list-item-meta-title"
              >
                <div
                  className="pro-components-notice-icon-notice-list-title"
                >
                  You received 14 new weekly newspapers
                </div>
              </h4>
              <div
                className="list-item-meta-description"
              >
                <div>
                  <div
                    className="pro-components-notice-icon-notice-list-description"
                  />
                  <div
                    className="pro-components-notice-icon-notice-list-datetime"
                  >
                    10 months ago
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="border-left-0 border-right-0 rounded-0 list-item pro-components-notice-icon-notice-list-item  list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <div
            className="list-item-meta pro-components-notice-icon-notice-list-meta"
          >
            <div
              className="list-item-meta-avatar"
            >
              <span
                className="avatar pro-components-notice-icon-notice-list-avatar avatar-circle avatar-image"
              >
                <img
                  alt=""
                  src="https://gw.alipayobjects.com/zos/rmsportal/GvqBnKhFgObvnSGkDsje.png"
                />
              </span>
            </div>
            <div
              className="list-item-meta-content"
            >
              <h4
                className="list-item-meta-title"
              >
                <div
                  className="pro-components-notice-icon-notice-list-title"
                >
                  You received 14 new weekly newspapers
                </div>
              </h4>
              <div
                className="list-item-meta-description"
              >
                <div>
                  <div
                    className="pro-components-notice-icon-notice-list-description"
                  />
                  <div
                    className="pro-components-notice-icon-notice-list-datetime"
                  >
                    10 months ago
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="border-left-0 border-right-0 rounded-0 list-item pro-components-notice-icon-notice-list-item  list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <div
            className="list-item-meta pro-components-notice-icon-notice-list-meta"
          >
            <div
              className="list-item-meta-avatar"
            >
              <span
                className="avatar pro-components-notice-icon-notice-list-avatar avatar-circle avatar-image"
              >
                <img
                  alt=""
                  src="https://gw.alipayobjects.com/zos/rmsportal/OKJXDXrmkNshAMvwtvhu.png"
                />
              </span>
            </div>
            <div
              className="list-item-meta-content"
            >
              <h4
                className="list-item-meta-title"
              >
                <div
                  className="pro-components-notice-icon-notice-list-title"
                >
                  You received 14 new weekly newspapers
                </div>
              </h4>
              <div
                className="list-item-meta-description"
              >
                <div>
                  <div
                    className="pro-components-notice-icon-notice-list-description"
                  />
                  <div
                    className="pro-components-notice-icon-notice-list-datetime"
                  >
                    10 months ago
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        className="border-0 rounded-0 clear-all-button pt-2 pb-2 shadow font-size-14 font-weight-bold btn btn-primary btn-block"
        disabled={false}
        type="button"
      >
        Clear All
      </button>
    </div>
    <div
      aria-hidden={true}
      aria-labelledby="uncontrolled-tab-example-tab-events"
      className="fade tab-pane"
      id="uncontrolled-tab-example-tabpane-events"
      role="tabpanel"
    >
      <div
        className="list-items shadow list-group"
        onKeyDown={[Function]}
      >
        <div
          className="border-left-0 border-right-0 rounded-0 list-item pro-components-notice-icon-notice-list-item border-top-0 list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <div
            className="list-item-meta pro-components-notice-icon-notice-list-meta"
          >
            <div
              className="list-item-meta-avatar"
            >
              <span
                className="avatar pro-components-notice-icon-notice-list-avatar avatar-circle avatar-image"
              >
                <img
                  alt=""
                  src="https://gw.alipayobjects.com/zos/rmsportal/ThXAXghbEsBCCSDihZxY.png"
                />
              </span>
            </div>
            <div
              className="list-item-meta-content"
            >
              <h4
                className="list-item-meta-title"
              >
                <div
                  className="pro-components-notice-icon-notice-list-title"
                >
                  All Events Posted
                </div>
              </h4>
              <div
                className="list-item-meta-description"
              >
                <div>
                  <div
                    className="pro-components-notice-icon-notice-list-description"
                  />
                  <div
                    className="pro-components-notice-icon-notice-list-datetime"
                  >
                    10 months ago
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="border-left-0 border-right-0 rounded-0 list-item pro-components-notice-icon-notice-list-item  list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <div
            className="list-item-meta pro-components-notice-icon-notice-list-meta"
          >
            <div
              className="list-item-meta-avatar"
            >
              <span
                className="avatar pro-components-notice-icon-notice-list-avatar avatar-circle avatar-image"
              >
                <img
                  alt=""
                  src="https://gw.alipayobjects.com/zos/rmsportal/ThXAXghbEsBCCSDihZxY.png"
                />
              </span>
            </div>
            <div
              className="list-item-meta-content"
            >
              <h4
                className="list-item-meta-title"
              >
                <div
                  className="pro-components-notice-icon-notice-list-title"
                >
                  React Event Posted
                </div>
              </h4>
              <div
                className="list-item-meta-description"
              >
                <div>
                  <div
                    className="pro-components-notice-icon-notice-list-description"
                  />
                  <div
                    className="pro-components-notice-icon-notice-list-datetime"
                  >
                    10 months ago
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="border-left-0 border-right-0 rounded-0 list-item pro-components-notice-icon-notice-list-item  list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <div
            className="list-item-meta pro-components-notice-icon-notice-list-meta"
          >
            <div
              className="list-item-meta-avatar"
            >
              <span
                className="avatar pro-components-notice-icon-notice-list-avatar avatar-circle avatar-image"
              >
                <img
                  alt=""
                  src="https://gw.alipayobjects.com/zos/rmsportal/OKJXDXrmkNshAMvwtvhu.png"
                />
              </span>
            </div>
            <div
              className="list-item-meta-content"
            >
              <h4
                className="list-item-meta-title"
              >
                <div
                  className="pro-components-notice-icon-notice-list-title"
                >
                  A new meetup at falana complex
                </div>
              </h4>
              <div
                className="list-item-meta-description"
              >
                <div>
                  <div
                    className="pro-components-notice-icon-notice-list-description"
                  />
                  <div
                    className="pro-components-notice-icon-notice-list-datetime"
                  >
                    10 months ago
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        className="border-0 rounded-0 clear-all-button pt-2 pb-2 shadow font-size-14 font-weight-bold btn btn-primary btn-block"
        disabled={false}
        type="button"
      >
        Clear All
      </button>
    </div>
    <div
      aria-hidden={true}
      aria-labelledby="uncontrolled-tab-example-tab-messages"
      className="fade tab-pane"
      id="uncontrolled-tab-example-tabpane-messages"
      role="tabpanel"
    >
      <div
        className="list-items shadow list-group"
        onKeyDown={[Function]}
      >
        <div
          className="border-left-0 border-right-0 rounded-0 list-item pro-components-notice-icon-notice-list-item border-top-0 list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <div
            className="list-item-meta pro-components-notice-icon-notice-list-meta"
          >
            <div
              className="list-item-meta-avatar"
            >
              <span
                className="avatar pro-components-notice-icon-notice-list-avatar avatar-circle avatar-image"
              >
                <img
                  alt=""
                  src="https://gw.alipayobjects.com/zos/rmsportal/ThXAXghbEsBCCSDihZxY.png"
                />
              </span>
            </div>
            <div
              className="list-item-meta-content"
            >
              <h4
                className="list-item-meta-title"
              >
                <div
                  className="pro-components-notice-icon-notice-list-title"
                >
                  You have received new message from Falana Dhikana
                </div>
              </h4>
              <div
                className="list-item-meta-description"
              >
                <div>
                  <div
                    className="pro-components-notice-icon-notice-list-description"
                  />
                  <div
                    className="pro-components-notice-icon-notice-list-datetime"
                  >
                    10 months ago
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="border-left-0 border-right-0 rounded-0 list-item pro-components-notice-icon-notice-list-item  list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <div
            className="list-item-meta pro-components-notice-icon-notice-list-meta"
          >
            <div
              className="list-item-meta-avatar"
            >
              <span
                className="avatar pro-components-notice-icon-notice-list-avatar avatar-circle avatar-image"
              >
                <img
                  alt=""
                  src="https://gw.alipayobjects.com/zos/rmsportal/ThXAXghbEsBCCSDihZxY.png"
                />
              </span>
            </div>
            <div
              className="list-item-meta-content"
            >
              <h4
                className="list-item-meta-title"
              >
                <div
                  className="pro-components-notice-icon-notice-list-title"
                >
                  You have received new message from Someone
                </div>
              </h4>
              <div
                className="list-item-meta-description"
              >
                <div>
                  <div
                    className="pro-components-notice-icon-notice-list-description"
                  />
                  <div
                    className="pro-components-notice-icon-notice-list-datetime"
                  >
                    10 months ago
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="border-left-0 border-right-0 rounded-0 list-item pro-components-notice-icon-notice-list-item  list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <div
            className="list-item-meta pro-components-notice-icon-notice-list-meta"
          >
            <div
              className="list-item-meta-avatar"
            >
              <span
                className="avatar pro-components-notice-icon-notice-list-avatar avatar-circle avatar-image"
              >
                <img
                  alt=""
                  src="https://gw.alipayobjects.com/zos/rmsportal/OKJXDXrmkNshAMvwtvhu.png"
                />
              </span>
            </div>
            <div
              className="list-item-meta-content"
            >
              <h4
                className="list-item-meta-title"
              >
                <div
                  className="pro-components-notice-icon-notice-list-title"
                >
                  Someone wants to connect with you
                </div>
              </h4>
              <div
                className="list-item-meta-description"
              >
                <div>
                  <div
                    className="pro-components-notice-icon-notice-list-description"
                  />
                  <div
                    className="pro-components-notice-icon-notice-list-datetime"
                  >
                    10 months ago
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        className="border-0 rounded-0 clear-all-button pt-2 pb-2 shadow font-size-14 font-weight-bold btn btn-primary btn-block"
        disabled={false}
        type="button"
      >
        Clear All
      </button>
    </div>
  </div>,
]
`;
